src/CDN/CloudFront.php

```diff
@@ -88,50 +102,56 @@ class CloudFront implements CDNInterface
     private $version;
 
     /**
-     * @todo: Make mandatory argument 5 and 6 when support for aws/aws-sdk-php < 3.0 is dropped.
+     * NEXT_MAJOR: Use `CloudFrontClient $client, string $distributionId, string $path` as signature for this method.
      *
-     * @param string $path
-     * @param string $key
-     * @param string $secret
-     * @param string $distributionId
+     * @param CloudFrontClient|string $clientOrPath
+     * @param string                  $distributionIdOrKey
+     * @param string                  $pathOrSecret
+     * @param string|null             $distributionId
      */
     public function __construct(
-        $path,
-        $key,
-        $secret,
-        $distributionId,
-        ?string $region = null,
-        ?string $version = null
+        $clientOrPath,
+        $distributionIdOrKey,
+        $pathOrSecret,
+        $distributionId = null
+        /* , ?string $region = null, ?string $version = null */
     ) {
-        $this->path = $path;
-        $this->key = $key;
-        $this->secret = $secret;
-        $this->distributionId = $distributionId;
-
-        // @todo: Remove the following conditional block when support for aws/aws-sdk-php < 3.0 is dropped.
-        if (class_exists(Sdk::class)) {
-            if (null === $region) {
-                throw new \TypeError(sprintf(
-                    'Argument 5 for "%s()" is required and can not be null when aws/aws-sdk-php >= 3.0 is installed.',
-                    __METHOD__
-                ));
-            }
-
-            if (null === $version) {
-                throw new \TypeError(sprintf(
-                    'Argument 6 for "%s()" is required and can not be null when aws/aws-sdk-php >= 3.0 is installed.',
+        // NEXT_MAJOR: Remove this check and the `else` block.
+        if ($clientOrPath instanceof CloudFrontClient) {
```

Comment:
Using an early return, can you make it so there is no `else`, and the `NEXT_MAJOR` is as easy as removing an `if (!$clientOrPath instanceof CloudFrontClient) {` code block? IMO we should make those instructions as easy as possible, and reduce the future git diffs to the minimum.